crates/interpreter/src/evaluate.rs

```diff
@@ -186,6 +186,63 @@ pub fn evaluate(emit: &EmitResult, global: Rc<RefCell<Object>>) -> Result<JSValu
                 stack.pop().ok_or(EvalError::EmptyStack)?;
             }
 
+            Opcode::Eq => {
+                let rval = stack.pop().ok_or(EvalError::EmptyStack)?;
+                let lval = stack.pop().ok_or(EvalError::EmptyStack)?;
+                // TODO: Add is special, i.e. string concat
+
+                match lval {
+                    JSValue::Number(lval) => stack.push(JSValue::Boolean(lval == to_number(&rval))),
+                    JSValue::String(_lval) => {
+                        // TODO: compare strings
+                        return Err(EvalError::NotImplemented(
+                            "compare strings with ==".to_owned(),
+                        ));
+                    }
+                    JSValue::Boolean(lval) => {
+                        stack.push(JSValue::Boolean(lval == to_boolean(&rval)))
+                    }
+                    _ => stack.push(JSValue::Boolean(to_boolean(&lval) == to_boolean(&rval))),
+                }
+            }
+
+            Opcode::Ne => {
+                let rval = stack.pop().ok_or(EvalError::EmptyStack)?;
+                let lval = stack.pop().ok_or(EvalError::EmptyStack)?;
+
+                match lval {
+                    JSValue::Number(lval) => stack.push(JSValue::Boolean(lval != to_number(&rval))),
+                    JSValue::String(_lval) => {
+                        // TODO: compare strings
+                        return Err(EvalError::NotImplemented(
+                            "compare strings with ==".to_owned(),
+                        ));
+                    }
+                    JSValue::Boolean(lval) => {
+                        stack.push(JSValue::Boolean(lval != to_boolean(&rval)))
+                    }
+                    _ => stack.push(JSValue::Boolean(to_boolean(&lval) != to_boolean(&rval))),
+                }
+            }
+
+            Opcode::StrictNe => {
+                let rval = stack.pop().ok_or(EvalError::EmptyStack)?;
```

Comment:
Move this to `Opcode::StrictEq` above. The implementation can be just `!strict_equality(lhs, rhs)`